common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1393,6 +1405,7 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
           // the client in cases of duplicate even though no data is written.
           if (isDuplicateBlock()) {
             freeDeferredBufs();
+            mergeManager.pushMergeMetrics.ignoredBlockBytes.mark(buf.limit());
             return;
           }
           abortIfNecessary();
```

Comment:
Even when the server aborts, it ignores the bytes.

Reply:
Make sense, have updated the PR to call `updateIgnoredBytesWithDeferredBufs()` in `abortIfNecessary()`.

Reply:
Here it should be `buf.remaining`

Reply:
Fixed.